service_config.go

```diff
@@ -229,15 +230,28 @@ type jsonName struct {
 	Method  *string
 }
 
-func (j jsonName) generatePath() (string, bool) {
+var (
+	errMissingService             = errors.New("'service' is missing")
+	errDuplicatedName             = errors.New("duplicated name")
+	errEmptyServiceNonEmptyMethod = errors.New("cannot combine empty 'service' and non-empty 'method'")
+)
+
+func (j jsonName) generatePath() (string, error) {
 	if j.Service == nil {
-		return "", false
+		return "", errMissingService
```

Comment:
I think there's some inconsistency in `service_config.proto`.
`service` shouldn't be marked as required, and [this](https://github.com/grpc/grpc-proto/blob/dd2dca318eb197b96b60f22297871fb1ed862800/grpc/service_config/service_config.proto#L61) line says `not present` is considered the same as `""`.

(Does the json library handle `null` correctly? Can you add a test? Thanks!)

`j.Service == nil` should be considered the same as `*j.Service == ""`, and we check if `method` is set, as in the following `if`.

---
On whether this is a breaking change, this only affect configs where service is not set, and method is set to non-empty string. This is not an valid config anyway, I think failing those would be OK.

Reply:
https://github.com/grpc/grpc-go/pull/3684/files#diff-656c5a93bda2544d9c7c7ea6f4a37af3R376
test is already there

Reply:
Oh, right, that's great!

Did you fix this
> j.Service == nil should be considered the same as *j.Service == "", and we check if method is set, as in the following if.

I saw you force pushed, but I can't find the fix for it.

Reply:
I think I finally got it, was a bit comfused. Please check out the latest commit